src/Chapter1.hs

```diff
@@ -553,8 +560,11 @@ value after "=" where the condition is true.
 
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z = do
+    let maxN = foldr1 max [x, y, z]
+    let minN = foldr1 min [x, y, z]
+    [n | n <- [x,y,z], n < maxN, n > minN]!!0
```

Comment:
Interesting solution 🤔 Nice that you have an understanding of folding and typeclasses. We will cover them later